### ISSUE TITLE
Polyfill of string.startsWith not compatible with Microsoft ASP.NET Web Forms

### ISSUE THREAD

**Issue by GarelliSwissRe** [formatjs/formatjs#4170]
## Which package?
icu-messageformat-parser

## Describe the bug
The file parser.ts at line 116 defines a function _startsWith_. The function detects whether a native _string.startsWith(search, position)_ exists and if it doesn't, it simulates with string.slice (IE11 scenario).
Unfortunately the Microsoft library MsAjax.js (https://ajax.aspnetcdn.com/ajax/4.0/1/MicrosoftAjaxCore.js) , part of MS WebForms, seems to introduce a polyfill with the following code

```
String.prototype.startsWith= function(n) {
  return this.,substr(0, n.length) === n
}
```

that defines a startWith not supporting the parameter _position_ and breaking the fortmatjs library.
While the problem depends on Microsoft non-compliant implementation, WebForms is a popular library and a work-around would be desirable. 
A slightly different check on the presence of a compliant native startsWith (e.g. test for the second parameter) might solve the problem.

The library ckeditor had a similar issue (not identical) as described in this issue https://github.com/ckeditor/ckeditor4-react/issues/76

## To Reproduce
The problem can be reproduced by importing Microsoft WebForms before importing formatjs. The issue is present with .Net Framework 4.7.2 (https://dotnet.microsoft.com/en-us/download/dotnet-framework/net472) but probably applies to all version.

The bundle the injects the problem is
```
<script src="[/bundles/MsAjaxJs?v=D6VN0fHlwFSIWjbVzi6mZyE9Ls-4LNrSSYVGRU46XF81](http://localhost:8081/bundles/MsAjaxJs?v=D6VN0fHlwFSIWjbVzi6mZyE9Ls-4LNrSSYVGRU46XF81)" type="text/javascript"></script>
```

Unfortunately WebForms uses a bundle mechanism that generates. The issue may be simulated by copying only the Ms polyfill without the need of the full library. Source asp code can be provided if useful.

```
<script>
String.prototype.startsWith= function(n) {
  return this.,substr(0, n.length) === n
}
</script>
```

## Expected behavior
The function startsWith in parser.ts should work even when Microsoft WebForms library is import before

## Desktop (please complete the following information):

- OS: [Windows]
- Browser [chrome]
- Version [120+]

**Comment by apolakipso** [comment]
This should be fixed with this merged PR: #4130

**Comment by longlho** [comment]
fixed
